pysmt/factory.py

```diff
@@ -259,6 +319,31 @@ def all_solvers(self, logic=None):
 
         return solvers
 
+    def all_unsat_core_solvers(self, logic=None):
```

Comment:
Same as above. We can factor this with all_solvers, by introducing an intermediate function that looks for a solver within a set.